packages/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx

```diff
@@ -167,7 +167,7 @@ const GridHeaderFilterCell = React.forwardRef<HTMLDivElement, GridHeaderFilterCe
 
     const applyFilterChanges = React.useCallback(
       (updatedItem: GridFilterItem) => {
-        if (item.value && !updatedItem.value) {
+        if (item.value && updatedItem.value === undefined) {
```

Comment:
```suggestion
        if (item.value && updatedItem.value == undefined) {
```

to allow checking for `null` as well ... just to be safe!

Reply:
thanks for the suggestion
`null`  values already handled here https://github.com/mui/mui-x/blob/8ee87b5689b772634f2ced89e83493a095b9a681/packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputBoolean.tsx#L26
I have also added test for it 2d3521d